dbms/src/Storages/MergeTree/MergeTreeBlockInputStream.h

```diff
@@ -48,9 +48,10 @@ class MergeTreeBlockInputStream : public MergeTreeBaseBlockInputStream
     bool getNewTask() override;
 
 private:
-    Block header;
+    mutable Block header;
```

Comment:
:-1: 

This is a red flag.
May lead to race conditions in `getHeader` method.